src/duc/cmd-json.c

```diff
@@ -0,0 +1,192 @@
+#include "config.h"
+
+#include <limits.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <ctype.h>
+
+#include "cmd.h"
+#include "duc.h"
+	
+
+static bool opt_apparent = false;
+static char *opt_database = NULL;
+static double opt_min_size = 0;
+static bool opt_exclude_files = false;
+
+
+static void indent(int n)
+{
+	int i;
+	for(i=0; i<n; i++) {
+		putchar(' ');
+	}
+}
+
+
+static void print_escaped(const char *s)
+{
+	while(*s) {
+		switch(*s) {
+			case '"': printf("\""); break;
+			case '\t': putchar('\t'); break;
+			case '\n': putchar('\n'); break;
+			case '\r': putchar('\r'); break;
+			default: putchar(*s); break;
+		}
+		s++;
+	}
+}
+
+
+static void dump(duc *duc, duc_dir *dir, int depth, off_t min_size, int ex_files)
+{
+	struct duc_dirent *e;
+	bool first = true;
+
+	duc_size_type st = opt_apparent ? DUC_SIZE_TYPE_APPARENT : DUC_SIZE_TYPE_ACTUAL;
+
+	while( (e = duc_dir_read(dir, DUC_SIZE_TYPE_ACTUAL, DUC_SORT_SIZE)) != NULL) {
+
+		off_t size = duc_get_size(&e->size, st);
+
+		if (!first) printf(",\n");
+
+		if(e->type == DUC_FILE_TYPE_DIR && size>= min_size) {
+			duc_dir *dir_child = duc_dir_openent(dir, e);
+			if(dir_child) {
+				indent(depth);
```

Comment:
At first, I tried to add a new method to merge the root print and the children prints. It was really painful due to commas, in the end I added a boolean to handle those. Thoughts welcome !